### ISSUE TITLE
Typeclass instance as axiom is treated differently by Coq 8.7.2 and 8.8.0

### ISSUE THREAD

**Issue by anton-trunov** [coq/coq#7874]
#### Version

Coq 8.8.0

#### Operating system

macOS 10.13.5

#### Description of the problem

```
From Coq Require Import Relations.
From Coq Require Import RelationClasses.

Section Foo.
Variable T : Type.
Variable equiv: relation T.

Axiom equivalence_equiv: Equivalence equiv.

Goal forall t : T, equiv t t.
Proof. reflexivity. Qed.  (* works with Coq 8.7.2, but does not with 8.8.0 *)

End Foo.
```

If I change
```
Axiom equivalence_equiv: Equivalence equiv.
```
into
```
Variable equivalence_equiv: Equivalence equiv.
```
it works with both 8.7.2 and 8.8.0.

I don't know what kind of behavior is desirable here. The issue is based on [this SO answer](https://stackoverflow.com/a/50946887/2747511).

**Comment by SkySkimmer** [comment]
This is intended. If you want an axiomatic instance use Declare Instance.

**Comment by anton-trunov** [comment]
@SkySkimmer Are you still supporting Coq 8.7.2? If so, then I think it should be reopened as a bug for Coq 8.7.2.

**Comment by SkySkimmer** [comment]
No, the behaviour changed for 8.8.

**Comment by anton-trunov** [comment]
I see, thanks.

**Comment by bollu** [comment]
@SkySkimmer could you clarify what the difference is between `Axiom` and `Variable`? I thought they were interchangeable, but were different for "documentation" / "math-like" purposes.

**Comment by SkySkimmer** [comment]
In a section Variable is a local assumption. Outside sections they're the same.
So in the example in the OP `Axiom equivalence_equiv: Equivalence equiv.` will become an axiom that every relation is an equivalence when the section is closed.

**Comment by bollu** [comment]
wow, okay. Could you tell me where this is specified in the reference manual?  Neither [the section mechanism](https://coq.inria.fr/refman/language/gallina-extensions.html#section-mechanism) nor [gallina definitions](https://coq.inria.fr/refman/language/gallina-specification-language.html#gallina-definitions) seems to record this interaction?  Sorry if I missed it.

**Comment by SkySkimmer** [comment]
Around here https://coq.inria.fr/refman/language/gallina-specification-language.html#coq:cmd.variable

**Comment by bollu** [comment]
@SkySkimmer I am still not sure what explicitly talks about this :) Could you show me the text?

**Comment by SkySkimmer** [comment]
>When the current section is closed, name ident will be unknown and every object using this variable will be explicitly parametrized (the variable is discharged).

**Comment by bollu** [comment]
Ah, OK. May I add an example about this pitfall to the doc? I find that sentence hard to parse, as a Coq non-expert :)

**Comment by Zimmi48** [comment]
For sure the documentation of sections should be improved and mention explicitly the duality `Variable` / `Parameter` and all the commands that can be used inside a section (like `Context` for instance).
So feel free to open a PR.
That being said, I'm not sure why you're using the word "pitfall" and you should notice that the use of `Variable` outside a section gives you a warning currently (though we should probably make this warning more explicit: `Variable` used to declare an axiom, maybe you meant to use `Parameter` instead).
